File: tests/bindings/dotnet/foo.Tests/EnumTest.cs
using System;
using Xunit;
using foo;

namespace foo.Tests
{
    public class EnumTests
    {
        [Fact]
        public void EnumZeroToFiveEchoTest()
        {
            var value = EnumZeroToFive.Zero;
            var result = EnumEchoFunctions.EnumZeroToFiveEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(0, (int)result);

            value = EnumZeroToFive.One;
            result = EnumEchoFunctions.EnumZeroToFiveEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(1, (int)result);

            value = EnumZeroToFive.Two;
            result = EnumEchoFunctions.EnumZeroToFiveEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(2, (int)result);

            value = EnumZeroToFive.Three;
            result = EnumEchoFunctions.EnumZeroToFiveEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(3, (int)result);

            value = EnumZeroToFive.Four;
            result = EnumEchoFunctions.EnumZeroToFiveEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(4, (int)result);

            value = EnumZeroToFive.Five;
            result = EnumEchoFunctions.EnumZeroToFiveEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(5, (int)result);
        }

        [Fact]
        public void EnumOneToSixEchoTest()
        {
            var value = EnumOneToSix.One;
            var result = EnumEchoFunctions.EnumOneToSixEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(1, (int)result);

            value = EnumOneToSix.Two;
            result = EnumEchoFunctions.EnumOneToSixEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(2, (int)result);

            value = EnumOneToSix.Three;
            result = EnumEchoFunctions.EnumOneToSixEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(3, (int)result);

            value = EnumOneToSix.Four;
            result = EnumEchoFunctions.EnumOneToSixEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(4, (int)result);

            value = EnumOneToSix.Five;
            result = EnumEchoFunctions.EnumOneToSixEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(5, (int)result);

            value = EnumOneToSix.Six;
            result = EnumEchoFunctions.EnumOneToSixEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(6, (int)result);
        }

        [Fact]
        public void EnumDisjointEchoTest()
        {
            var value = EnumDisjoint.Five;
            var result = EnumEchoFunctions.EnumDisjointEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(5, (int)result);

            value = EnumDisjoint.One;
            result = EnumEchoFunctions.EnumDisjointEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(1, (int)result);

            value = EnumDisjoint.Twenty;
            result = EnumEchoFunctions.EnumDisjointEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(20, (int)result);

            value = EnumDisjoint.Four;
            result = EnumEchoFunctions.EnumDisjointEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(4, (int)result);

            value = EnumDisjoint.Seven;
            result = EnumEchoFunctions.EnumDisjointEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(7, (int)result);

            value = EnumDisjoint.Two;
            result = EnumEchoFunctions.EnumDisjointEcho(value);
            Assert.Equal(value, result);
            Assert.Equal(2, (int)result);
        }
    }
}

File: tests/bindings/java/foo-tests/src/test/java/io/stepfunc/foo_test/EnumTest.java
package io.stepfunc.foo_test;

import io.stepfunc.foo.*;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class EnumTest {
    @Test
    public void EnumZeroToFiveEchoTest() {
        EnumZeroToFive value = EnumZeroToFive.ZERO;
        EnumZeroToFive result = EnumEchoFunctions.enumZeroToFiveEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumZeroToFive.ONE;
        result = EnumEchoFunctions.enumZeroToFiveEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumZeroToFive.TWO;
        result = EnumEchoFunctions.enumZeroToFiveEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumZeroToFive.THREE;
        result = EnumEchoFunctions.enumZeroToFiveEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumZeroToFive.FOUR;
        result = EnumEchoFunctions.enumZeroToFiveEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumZeroToFive.FIVE;
        result = EnumEchoFunctions.enumZeroToFiveEcho(value);
        assertThat(result).isEqualTo(value);
    }

    @Test
    public void EnumOneToSixEchoTest() {
        EnumOneToSix value = EnumOneToSix.ONE;
        EnumOneToSix result = EnumEchoFunctions.enumOneToSixEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumOneToSix.TWO;
        result = EnumEchoFunctions.enumOneToSixEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumOneToSix.THREE;
        result = EnumEchoFunctions.enumOneToSixEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumOneToSix.FOUR;
        result = EnumEchoFunctions.enumOneToSixEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumOneToSix.FIVE;
        result = EnumEchoFunctions.enumOneToSixEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumOneToSix.SIX;
        result = EnumEchoFunctions.enumOneToSixEcho(value);
        assertThat(result).isEqualTo(value);
    }

    @Test
    public void EnumDisjointEchoTest() {
        EnumDisjoint value = EnumDisjoint.FIVE;
        EnumDisjoint result = EnumEchoFunctions.enumDisjointEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumDisjoint.ONE;
        result = EnumEchoFunctions.enumDisjointEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumDisjoint.TWENTY;
        result = EnumEchoFunctions.enumDisjointEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumDisjoint.FOUR;
        result = EnumEchoFunctions.enumDisjointEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumDisjoint.SEVEN;
        result = EnumEchoFunctions.enumDisjointEcho(value);
        assertThat(result).isEqualTo(value);

        value = EnumDisjoint.TWO;
        result = EnumEchoFunctions.enumDisjointEcho(value);
        assertThat(result).isEqualTo(value);
    }

    @Test
    public void EnumSingleEchoTest() {
        EnumSingle value = EnumSingle.SINGLE;
        EnumSingle result = EnumEchoFunctions.enumSingleEcho(value);
        assertThat(result).isEqualTo(value);
    }
}
